preliz/ppls/pymc_io.py

```diff
@@ -210,12 +211,40 @@ def non_constant_parents(var_, free_rvs):
     return parents
 
 
-def parse_backfitting(model, data):
-    posterior = az.extract(data, group="posterior")
+def parse_backfitting(model, data, alternative=None):
+    """
+    Get the model information for fitting the samples from prior into user provided model's prior.
+    We need to "backfit" because we can not use arbitrary samples as priors.
 
-    model_info = get_model_information(model)[2]
-    parsed_info = [(dist, var) for var, dist in model_info.items()]
+    Parameters
+    ----------
+    model : A PyMC model
+    A probabilistic model
+
+    data : np.ndarray
+    Samples which needs to be fitted using "backfitting"
+
+    alternative : list, defaults to None
+    Users can add the model variables to consider alternative distributions while fitting samples
 
+    """
+
+    data = np.asarray(data)
+    with model:
+        idata = sample(random_seed=4124)
+    posterior = az.extract(idata, group="posterior")
+    model_info = get_model_information(model)[2]
+    if alternative:
+        parsed_info = []
+        for key in model_info.keys():
+            if key in alternative:
+                dists_extra = [pz.Gamma(), pz.Normal(), pz.HalfNormal(), model_info[key]]
+                idx = np.argmin(fit_to_sample(dists_extra, data, data.min(), data.max()).losses)
```

Comment:
Better to use pz.mle function that takes care of this

Reply:
Yes, pz.mle seems to be the best option